runtime/str.go

```diff
@@ -205,6 +205,45 @@ func strEq(f *Frame, v, w *Object) (*Object, *BaseException) {
 	return strCompare(v, w, False, True, False), nil
 }
 
+func strFind(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	// TODO: Support finding index of unicode substring.
+	expectedTypes := []*Type{StrType, StrType, ObjectType, ObjectType}
+	argc := len(args)
+	if argc == 2 || argc == 3 {
+		expectedTypes = expectedTypes[:argc]
+	}
+	if raised := checkMethodArgs(f, "find/index", args, expectedTypes...); raised != nil {
+		return nil, raised
+	}
+	s := toStrUnsafe(args[0]).Value()
+	l := len(s)
+	start, end := 0, l
+	if argc >= 3 {
+		startParsed, raised := adjustBounds(f, args[2], l, false)
+		if raised != nil {
+			return nil, raised
+		}
+		start = startParsed
+	}
+	if argc == 4 {
+		endParsed, raised := adjustBounds(f, args[3], l, true)
+		if raised != nil {
+			return nil, raised
+		}
+		end = endParsed
+	}
+	if start > end {
+		return NewInt(-1).ToObject(), nil
+	}
+	sep := toStrUnsafe(args[1]).Value()
+	s = s[start:end]
```

Comment:
It's better to avoid reassigning to local variables when possible since it's less obvious to readers what value it has at what point in the function. You can just pass s[start:end] directly to strings.Index() below.